gadgets/weather/src/lib.rs

```diff
@@ -38,7 +38,7 @@ impl Gadget for WeatherGadget {
                 match weather_data {
                     Ok(weather_data) => *weather_data_lock.locked() = weather_data,
                     Err(error) => {
-                        println!("Failed to retrieve the weather data from wttr.in: {error}")
+                        println!("Failed to retrieve the weather data from wttr.in: {error}");
```

Comment:
I wonder if we should enable `clippy::semicolon_if_nothing_returned`, it'd catch cases like this

Reply:
Personally i use clippy::pedantic, but if that's not something you'd want to do in the project, enabling it would be fine (or not even needed, it's not set to warn by default, as it's more of a pedantic thing :P)

Reply:
I personally don't really like clippy::pedantic, it feels too... pedantic, I'm more of a fan of enabling the warnings I like manually (but that gets a lot harder in a multicrate project like this)